### PR TITLE
Prefer same address when establishing streaming connections

### PR DESCRIPTION
**Pull request by JakaMohorkoDS**
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] Pull request title reflects its content
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

# Description:

- Streaming now uses the address used by configuration when establishing connections.
- Refactored device creation code in module manager

### CODE REVIEW

File: core/opendaq/modulemanager/src/module_manager_impl.cpp [hunk 1]
```diff
@@ -910,9 +791,90 @@ uint16_t ModuleManagerImpl::getServerCapabilityPriority(const ServerCapabilityPt
...
+    {
+        const auto priority = getServerCapabilityPriority(capability);
+        if (priority  > selectedPriority)
+        {
+            selectedCapability = capability;
+            selectedPriority = priority;
+        }
+    }
+
+    if (selectedCapability.assigned())
```
Comment (denise-opendaq) [inline core/opendaq/modulemanager/src/module_manager_impl.cpp:848]: do we have a case where selectedCapability is not assigned?

File: core/opendaq/modulemanager/src/module_manager_impl.cpp [hunk 2]
```diff
@@ -726,6 +579,34 @@ void ModuleManagerImpl::populateDeviceConfigFromConnStrOptions(const PropertyObj
     }
 }
 
+PropertyObjectPtr ModuleManagerImpl::populateDeviceConfig(const PropertyObjectPtr& config,
+                                                          const DeviceTypePtr& deviceType,
+                                                          const tsl::ordered_map<std::string, ObjectPtr<IBaseObject>>& connStrOptions)
+{
+    PropertyObjectPtr configClone;
+    if (config.assigned())
+        config.asPtr<IPropertyObjectInternal>()->clone(&configClone);
```
Comment (denise-opendaq) [inline core/opendaq/modulemanager/src/module_manager_impl.cpp:588]: configClone could be not assigned

File: core/opendaq/modulemanager/src/module_manager_impl.cpp [hunk 3]
```diff
@@ -428,215 +428,68 @@ ErrCode ModuleManagerImpl::createDevice(IDevice** device, IString* connectionStr
...
-                        configureStreamings(mirroredDeviceConfigPtr, configPtr);
-                        return OPENDAQ_SUCCESS;
-                    });
-            }
+            return err;
         }
-
-        return errCode;
+    }
+    catch (const DaqException& e)
```
Comment (denise-opendaq) [inline core/opendaq/modulemanager/src/module_manager_impl.cpp:482]: the same as in the daqTry

Comment (JakaMohorkoDS) [reply core/opendaq/modulemanager/src/module_manager_impl.cpp:482]: Yes, might just be a bit of a stylistic preference from my side, but I prefer actual try/catch statements when encasing such large code blocks.

**Review by denise-opendaq — APPROVED** [review]

**Pull request merged**